src/DocumentDB.ChangeFeedProcessor/ChangeFeedProcessorBuilder.cs

```diff
@@ -504,10 +500,14 @@ private string GetLeasePrefix()
 
         private async Task InitializeCollectionPropertiesForBuildAsync()
         {
-            this.feedDocumentClient = this.feedDocumentClient ?? this.feedCollectionLocation.CreateDocumentClient();
             this.changeFeedProcessorOptions = this.changeFeedProcessorOptions ?? new ChangeFeedProcessorOptions();
             this.databaseResourceId = this.databaseResourceId ?? await GetDatabaseResourceIdAsync(this.feedDocumentClient, this.feedCollectionLocation).ConfigureAwait(false);
             this.collectionResourceId = this.collectionResourceId ?? await GetCollectionResourceIdAsync(this.feedDocumentClient, this.feedCollectionLocation).ConfigureAwait(false);
         }
+
+        private void InitializeFeedDocumentClient()
+        {
+            this.feedDocumentClient = this.feedDocumentClient = new ChangeFeedDocumentClientHealthDecorator(this.feedDocumentClient ?? this.feedCollectionLocation.CreateDocumentClient(), this.healthMonitor, this.changeFeedProcessorOptions.ChangeFeedClientHealthOptions);
```

Comment:
remove one of "this.feedDocumentClient =" from "this.feedDocumentClient = this.feedDocumentClient =".